src/dialogs/more-info/ha-more-info-dialog.ts

```diff
@@ -0,0 +1,298 @@
+import "@material/mwc-button";
+import "@polymer/app-layout/app-toolbar/app-toolbar";
+import "@polymer/paper-dialog-scrollable/paper-dialog-scrollable";
+import "../../components/ha-icon-button";
+import "../../components/ha-dialog";
+import { isComponentLoaded } from "../../common/config/is_component_loaded";
+import { DOMAINS_MORE_INFO_NO_HISTORY } from "../../common/const";
+import { computeStateName } from "../../common/entity/compute_state_name";
+import { navigate } from "../../common/navigate";
+import "../../components/state-history-charts";
+import { removeEntityRegistryEntry } from "../../data/entity_registry";
+import { showEntityEditorDialog } from "../../panels/config/entities/show-dialog-entity-editor";
+import "../../state-summary/state-card-content";
+import { showConfirmationDialog } from "../generic/show-dialog-box";
+import "./more-info-content";
+import { customElement, LitElement, property, css } from "lit-element";
+import { html } from "lit-html";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import { getRecentWithCache, CacheConfig } from "../../data/cached-history";
+import { computeDomain } from "../../common/entity/compute_domain";
+import { mdiClose, mdiSettings, mdiPencil } from "@mdi/js";
+
+const DOMAINS_NO_INFO = ["camera", "configurator", "history_graph"];
+const EDITABLE_DOMAINS_WITH_ID = ["scene", "automation"];
+const EDITABLE_DOMAINS = ["script"];
+
+@customElement("ha-more-info-dialog")
+export class MoreInfoDialog extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() private _stateHistory?: object;
+
+  @property({ type: Boolean, reflect: true }) public large = false;
+
+  @property() private _cacheConfig?: CacheConfig;
+
+  private _interval?: number;
+
+  protected updated(changedProperties) {
+    super.updated(changedProperties);
+    if (changedProperties.has("hass")) {
+      const oldHass = changedProperties.get("hass");
+      if (!oldHass || oldHass.moreInfoEntityId !== this.hass.moreInfoEntityId) {
+        if (this.hass.moreInfoEntityId) {
+          this.setAttribute(
+            "data-domain",
+            computeDomain(this.hass.moreInfoEntityId)
+          );
+          this.large = false;
+          this._stateHistory = undefined;
+          if (this._computeShowHistoryComponent(this.hass.moreInfoEntityId)) {
+            this._cacheConfig = {
```

Comment:
Feels weird to have this be an object. We can just create this object inside the refreshHistoryMethod ?